### ISSUE TITLE
Allow noBib within edi invoice management

### ISSUE THREAD

**Issue by dlrueda** [sul-dlss/libsys-webforms#523]
User chooses "Allow a noBib" from EDIFACT invoice management menu

User enters three values to identify invoice line to update:
invoice number (doc_num in edi_lin table)
vendor ID (vend_id in edi_lin table)
invoice line number (edi_lin_num in edi_lin table)

Identify this entry in edi_lin table, get vend_unique_id value.  SQL currently used is in comment below.

Check if this vend_unique_id value exists in edi_sumrz_bib table, id001 column.  SQL for this is below.  If it exists already in edi_sumrz_bib table, return error message to user:
Flash message : "Cannot set this invoice line to noBib"
Error message that remains on screen: "This invoice line has a bib match in Symphony. It shouldn't need to be set to "allow noBib" : 
uniq_bib_id = _display edi_sumrz_bib.id001_
edi_ckey = _display edi_sumrz_bib.edi_ckey_ "

If vend_unique_id DOES NOT exist in edi_sumrz_bib table, then two tables will be updated in order to allow this invoice line to get processed as a noBib:
**edi_lin.vend_unique_id**
update this value to be vend_unique_id || "noBib"
SQL currently used is in comment below

**edi_sumrz_bib**
Will be writing a new row to this table with this vend_unique_id || 'noBib' and fake edi_ckey value.  This is what will allow this invoice line through as a noBib.

First need to figure out what edi_ckey value should be.  Currently code looks for minimum edi_sumrz_bib.edi_ckey value that is < 0, subtracts 1 from it, and assigns this new negative number to be edi_ckey for new row.  Rails code needs to do whatever steps to assign next negative value for edi_ckey.

Then insert new row to table.  SQL currently used is in comment below.

**Comment by dlrueda** [comment]
SQL currently used to retrieve this entry from edi_lin table, get vend_unique_id for this entry:
```
  SELECT vend_unique_id
    FROM edi_lin
   WHERE UPPER(vend_id) = UPPER(p_edi_vend_id)
     AND UPPER(doc_num) = UPPER(p_edi_doc_num)
     AND edi_lin_num = p_edi_line_num
     AND edi_sublin_count = 0;
```

**Comment by dlrueda** [comment]
SQL currently used to check if there is an entry in edi_sumrz_bib table matching this vend_unique_id value:
```
SELECT id001, edi_ckey
    FROM edi_sumrz_bib
   WHERE UPPER(id001) = UPPER(vend_unique_id);
```

**Comment by dlrueda** [comment]
example SQL to update edi_lin with "noBiB" appended to vend_unique_id to allow noBib processing:
```
UPDATE edi_lin
SET vend_unique_id = vend_unique_id || 'noBib'
   WHERE UPPER(vend_id) = UPPER(p_edi_vend_id)
     AND UPPER(doc_num) = UPPER(p_edi_doc_num)
     AND edi_lin_num = p_edi_line_num
     AND edi_sublin_count = 0;
```

**Comment by dlrueda** [comment]
SQL currently used to add new row to edi_sumrz_bib table

```
INSERT INTO edi_sumrz_bib
        (vend_code,
         id001,
         edi_ckey,
         load_date,
         active_record)
  VALUES(
        UPPER(p_edi_vend_id),
        vend_unique_id || 'noBib',
        fake_edi_ckey,
        SYSDATE,
        NULL);
```

**Comment by jgreben** [comment]
Fixed by #534

**Comment by dlrueda** [comment]
Acceptance criteria
- [x] Allows user to input vendor, invoice#, and line#
- [x] Rejects if vend_uniq_id for that vendor/invoice/line already in edi_sumrz_bib table
- [ ] If accepted, appends "noBib" to vend_uniq_id in edi_lin table for vendor/invoice/line
- [ ] If accepted, adds line to edi_sumrz_bib with vend_uniq_id"noBib"
- [ ] If accepted, assigns negative value (one "less" than current minimum) to edi_ckey value in edi_sumrz_bib
